src/core/border_router/routing_manager.cpp

```diff
@@ -1831,6 +1851,96 @@ void RoutingManager::DiscoveredPrefixTable::SignalTableChanged(void)
     mSignalTask.Post();
 }
 
+void RoutingManager::DiscoveredPrefixTable::ProcessNeighborAdvertMessage(
+    const Ip6::Nd::NeighborAdvertMessage &aNaMessage,
+    const Ip6::Address &                  aSrcAddress)
+{
+    Router *router = mRouters.FindMatching(aSrcAddress);
+
+    VerifyOrExit(router != nullptr);
+    VerifyOrExit(aNaMessage.IsRouterFlagSet());
+    UpdateRouterOnRx(*router);
+
+exit:
+    return;
+}
+
+void RoutingManager::DiscoveredPrefixTable::UpdateRouterOnRx(Router &aRouter)
+{
+    aRouter.mState   = Router::kActive;
+    aRouter.mTimeout = TimerMilli::GetNow() + Random::NonCrypto::AddJitter(Router::kActiveTimout, Router::kJitter);
+
+    mRouterTimer.FireAtIfEarlier(aRouter.mTimeout);
+}
+
+void RoutingManager::DiscoveredPrefixTable::HandleRouterTimer(Timer &aTimer)
+{
+    aTimer.Get<RoutingManager>().mDiscoveredPrefixTable.HandleRouterTimer();
+}
+
+void RoutingManager::DiscoveredPrefixTable::HandleRouterTimer(void)
+{
+    TimeMilli now      = TimerMilli::GetNow();
+    TimeMilli nextTime = now.GetDistantFuture();
+
+    for (Router &router : mRouters)
+    {
+        if (router.mState == Router::kStale)
+        {
+            continue;
+        }
+
+        if (router.mTimeout <= now)
+        {
+            if (router.mState == Router::kActive)
+            {
+                router.mState        = Router::kProbing;
+                router.mNsProbeCount = 0;
```

Comment:
nit: Can we use `mNsProbeCount > 0` to indicate a probing Router?
`kProbing` state seems unnecessary.

Reply:
Good suggestion. Thanks. I changed it in the new push. 
I agree the logic is simple enough that we can use the `mNsProbeCount` to represent the different states and code should be still easy to follow.